src/scripts/main.js

```diff
@@ -1 +1,96 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    document.addEventListener('contextmenu', () => {
+      resolve('Third promise was resolved');
+    });
+  });
```

Comment:
```suggestion
  let rightClick = false;
  let leftClick = false;

  document.addEventListener('mousedown', (e) => {
    switch (e.button) {
      case 0:
        leftClick = true;
        break;
      case 2:
        rightClick = true;
        break;
    }

    if (leftClick && rightClick) {
      resolve('Third promise was resolved');
    }
  });
```
Зараз все ще не працює, як потрібно.
Я навів приклад, як має бути - можна опрацьовувати event mousedown. В нього є властивість button. 0 - це ліва кнопка, 2 - права.
Можна відслідковувати, чи був клік на тій і на тій кнопці, і якщо так - тоді резолвити проміс